### ISSUE TITLE
util-linux su requires PAM

### ISSUE THREAD

**Issue by thesamesam** [shadow-maint/shadow#464]
Hi,

It looks like util-linux's implementation of `su` [requires](https://github.com/util-linux/util-linux/blob/161d3fd6c62167d3cd5f6e255648ba7391c28eb4/configure.ac#L2175) PAM. This doesn't personally bother me but in Gentoo, at present, we allow building systems without PAM. shadow's `su`, however, does support standalone usage.

I wonder if it'd be possible to keep shadow's `su` on life-support (but deprecated) for those who don't want to/can't use PAM?

**Comment by hallyn** [comment]
Thanks for pointing that out.  That may be a good enough reason to keep it.

**Comment by arachsys** [comment]
FWIW, I'd be a bit stuffed without su from shadow for the same reason: I maintain a distro which doesn't use PAM, so the util-linux su refuses to build.

**Comment by hallyn** [comment]
Thanks for the input @arachsys .

**Comment by hallyn** [comment]
I'm not sure where to best announce this - but we won't drop su unless and until there is an alternative.  My goal is to minimize community effort spent on redundant complex software, but I feel shadow has a responsibility to non-standard distros here.

I will however encourage those who can to switch to util-linux's su.  Most already had.

**Comment by arachsys** [comment]
Thanks, this is very much appreciated. As far as I know, shadow's implementation really is the only chfn/chsh/login/su option for people who (for example) have a statically-linked variant build of their distro which can't dlopen pam modules.

Are there any outstanding problems or design concerns with the current shadow su that you'd like someone to step up and fix? I'm happy to work on these if so.

**Comment by firasuke** [comment]
Any updates to this? Has `shadow` dropped `su`?

Also should distributions be switching to use the `util-linux` version of `su` (which requires PAM)? What if PAM is non-existent on said distributions, which version of `su` can be used?

**Comment by alejandro-colomar** [comment]
> Hi,
> 
> It looks like util-linux's implementation of `su` [requires](https://github.com/util-linux/util-linux/blob/161d3fd6c62167d3cd5f6e255648ba7391c28eb4/configure.ac#L2175) PAM. This doesn't personally bother me but in Gentoo, at present, we allow building systems without PAM. shadow's `su`, however, does support standalone usage.
> 
> I wonder if it'd be possible to keep shadow's `su` on life-support (but deprecated) for those who don't want to/can't use PAM?

@thesamesam, @firasuke, @arachsys   No, shadow's su(1) isn't going to be on life support.  It will have full support, and won't be deprecated, as far as I'm concerned.  I hope this is good news for you.  :)
